django/db/backends/mysql/features.py

```diff
@@ -50,6 +50,52 @@ class DatabaseFeatures(BaseDatabaseFeatures):
         'swedish_ci': 'utf8_swedish_ci',
     }
 
+    @cached_property
+    def django_test_skips(self):
+        skips = {
+            "This doesn't work on MySQL.": {
+                'db_functions.comparison.test_greatest.GreatestTests.test_coalesce_workaround',
+                'db_functions.comparison.test_least.LeastTests.test_coalesce_workaround',
+            },
+            'Running on MySQL requires utf8mb4 encoding (#18392).': {
+                'model_fields.test_textfield.TextFieldTests.test_emoji',
+                'model_fields.test_charfield.TestCharField.test_emoji',
+            },
+        }
+        if 'ONLY_FULL_GROUP_BY' in self.connection.sql_mode:
+            skips.update({
+                'GROUP BY optimization does not work properly when '
+                'ONLY_FULL_GROUP_BY mode is enabled on MySQL, see #31331.': {
+                    'aggregation.tests.AggregateTestCase.test_aggregation_subquery_annotation_multivalued',
+                    'annotations.tests.NonAggregateAnnotationTestCase.test_annotation_aggregate_with_m2o',
+                },
+            })
+        if (
+            self.connection.mysql_is_mariadb and
+            (10, 4, 3) < self.connection.mysql_version < (10, 5, 2)
+        ):
+            skips.update({
+                'https://jira.mariadb.org/browse/MDEV-19598': {
+                    'schema.tests.SchemaTests.test_alter_not_unique_field_to_primary_key',
+                },
+            })
+        if (
+            self.connection.mysql_is_mariadb and
+            (10, 4, 12) < self.connection.mysql_version < (10, 5)
+        ):
+            skips.update({
+                'https://jira.mariadb.org/browse/MDEV-22775': {
+                    'schema.tests.SchemaTests.test_alter_pk_with_self_referential_field',
+                },
+            })
+        if not self.supports_explain_analyze:
+            skips.update({
+                'MariaDB and MySQL >= 8.0.18 specific.': {
+                    'queries.test_explain.ExplainTests.test_mysql_analyze',
+                },
+            })
```

Comment:
We cannot move these skips, because that will cause running queries before a test database is created, see 493b26bbfc9cc0ad223ece131741cba2312ced0f (ticket-31888).

Reply:
Based on this and other cases, I think we should instead try to mark skips after the database is created (https://github.com/django/django/pull/13717#issuecomment-739609140).

Reply:
I moved `mark_expected_failures_and_skips` function call to the end of `db.backends.base.creation.create_test_db`.
Also, I moved `servers.test_liveserverthread.LiveServerThreadTest.test_closes_connections` to skip tests of sqlite3

Reply:
Assuming this works, don't forget that we'll need some solution (like the environment variable used before) to only run `mark_expected_failures_and_skips()` when running Django's test suite.

Reply:
I added the environment variable